docs/specification.md

```diff
@@ -1,150 +1,371 @@
+    """Convert a dict into a canonical string representation"""
+    return json.dumps(
+        item, separators=(",", ":"), ensure_ascii=False, allow_nan=False, sort_keys=True
+    )
+```
+
+This will turn the values into canonicalized string representations of the list objects, so lengths becomes the string `[248956422,133797422,135086622]`, names becomes the string `["chr1","chr2","chr3"]`, and sequences becomes the string 
+
+```
+["2648ae1bacce4ec4b6cf337dcae37816","907112d17fcb73bcab1ed1c72b97ce68","1511375dc2dd1b633af8cf439ae90cec"]
+```
+
+#### Step 3: Digest each canonicalized attribute value using the GA4GH digest algorithm.
+
+Apply the GA4GH digest algorithm to each attribute value. The GA4GH digest algorithm is described in detail in *Footnote F5*. This converts the value of each attribute in the seqcol into a digest string. Applying this to each value will produce a structure that looks like this:
+
+```json
+{
+  "lengths": "20e95aade8e72d399dbf7f82a9e84ba5cc4047dc8d791d62",
+  "names": "834e2529dc6262d1b774e19e502e4074a1227f0eb91b45a9",
+  "sequences": "78f45f5aa3b36a2a8fe1eec415258a036b3753f69acf05df"
+}
+```
+
+#### Step 4: Apply RFC-8785 again to canonicalize the JSON of new seqcol object representation.
+
+Here, we repeat step 2, except instead of applying to each value separately, we apply to the entire object. This will result in a canonical string representation of the object, which is this string:
+
+```
+{"lengths":"20e95aade8e72d399dbf7f82a9e84ba5cc4047dc8d791d62","names":"834e2529dc6262d1b774e19e502e4074a1227f0eb91b45a9","sequences":"78f45f5aa3b36a2a8fe1eec415258a036b3753f69acf05df"}
+```
+
+#### Step 5: Digest the final canonical representation again.
+
+Again using the same approach as in step 3, we now apply the GA4GH digest algorithm to this string. The result is the final unique identifier for this sequence collection:
+
+```
+64ff00b85402a4dc821752e1e8d56d3ecc4e29b55a930748
+```
+
+---
+
+### 2. API: A server RESTful API specification for retrieving and comparing sequence collections.
+
+The API has 3 top-level endpoints, for 3 functions:
+
+1. `/service-info`, for describing information about the service;
+2. `/collection`, for retrieving sequence collections; and
+3. `/comparison`, for comparing two sequence collections.
+
+Under these umbrella endpoints are a few more specific sub-endpoints, described in detail below:
+
+#### 2.1 Service info 
+- *Endpoint*: `GET /service-info` (`REQUIRED`)
+- *Description*: The service info endpoint provides information about the service
+- *Return value*: Must include the Seqcol schema that this server uses.
+
+The `/service-info` endpoint should follow the [GA4GH-wide specification for service info](https://github.com/ga4gh-discovery/ga4gh-service-info/) for general description of the service.
+Then, it should also add a few specific pieces of information under a `seqcol` property:
+ - `schema`: should return the JSON-schema implemented by the server.
```

Comment:
Would recommend in a later version to include a fully formed payload example

Reply:
agreed